mobsfscan/formatters/sarif.py

```diff
@@ -105,21 +106,41 @@ def create_result(path, rule_id, issue_dict, rules, rule_indices):
         rules[rule_id] = rule
         rule_indices[rule_id] = rule_index
 
-    locations = []
-    for item in issue_dict.get('files', []):
-        physical_location = om.PhysicalLocation(
-            artifact_location=om.ArtifactLocation(
-                uri=to_uri(item['file_path'])),
-        )
-        physical_location.region = om.Region(
-            start_line=item['match_lines'][0],
-            end_line=item['match_lines'][1],
-            start_column=item['match_position'][0],
-            end_column=item['match_position'][1],
-            snippet=om.ArtifactContent(text=item['match_string']),
-        )
-        locations.append(om.Location(physical_location=physical_location))
-    if not locations:
+    files = issue_dict.get('files', [])
+
+    # if there are locations - we iterate over them and create
+    # a separete result for each location
+    if files:
+        for item in files:
+            locations = []
```

Comment:
Should this be in the loop?
It will be recreated for each item, which doesn't look like what you want (or had before)

Reply:
Yes, it should be in the loop.

While "locations" is an array, it should normally only have one element, as per the SARIF spec:

>The locations array SHALL NOT be used to specify distinct occurrences of the same result which can be corrected independently.

Before - this was implemented incorrectly and all locations were added to the array of a single result. Now for each finding it produces a separate result with only one location in the array.